service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -97,52 +132,65 @@ public PolicyEvaluationResult evalPolicy(final PolicyEvaluationRequestV1 request
             } else {
                 supplementalSubjectAttributes = new HashSet<>(request.getSubjectAttributes());
             }
-            result = evalPolicy(request.getResourceIdentifier(), request.getSubjectIdentifier(), request.getAction(),
-                    supplementalResourceAttributes, supplementalSubjectAttributes);
+
+            for (PolicySet policySet : filteredPolicySets) {
+                result = evalPolicySet(policySet, subjectIdentifier, uri, action, supplementalResourceAttributes,
+                        supplementalSubjectAttributes);
+                if (result.getEffect() != Effect.NOT_APPLICABLE) {
+                    break;
+                }
+                // Continue to the next policy set if evaluation result is NOT_APPLICABLE
+            }
+
+            LOGGER.info(
+                    String.format(
+                            "Processed Policy Evaluation for: "
+                                    + "resourceUri=[%s], subjectIdentifier=[%s], action=[%s]," + " result=[%s]",
+                            uri, subjectIdentifier, action, result.getEffect()));
             this.cache.set(key, result);
         }
         return result;
     }
 
-    @Override
-    public PolicyEvaluationResult evalPolicy(final String uri, final String subjectIdentifier, final String action,
-            final Set<Attribute> supplementalResourceAttributes, final Set<Attribute> supplementalSubjectAttributes) {
-
-        if (uri == null || subjectIdentifier == null || action == null) {
+    List<PolicySet> filterPolicySetsByPriority(final String subjectIdentifier, final String uri,
+            final List<PolicySet> allPolicySets, final List<String> policySetsEvaluationOrder)
+            throws IllegalArgumentException {
 
-            LOGGER.error(
-                    String.format(
-                            "PolicyEvaluationResult input paramters cannot be null, "
-                                    + "resourceURI=[%s] subjectIdentifier=[%s] action=[%s]",
-                            uri, subjectIdentifier, action));
-
-            throw new IllegalArgumentException(
-                    "ACS Internal Error: PolicyEvaluationResult input paramters cannot be null.");
+        if (policySetsEvaluationOrder.isEmpty()) {
+            if (allPolicySets.size() > 1) {
+                LOGGER.error("Found more than one policy set during policy evaluation and no priority is provided. "
+                        + "Subject: " + subjectIdentifier + ", Resource: " + uri);
```

Comment:
Here we don't even use a delimiter in the message. Once again, can we just pick a convention and maintain consistency.